src/parquet/arrow/schema.h

```diff
@@ -44,13 +45,17 @@ ::arrow::Status PARQUET_EXPORT NodeToField(
 /// \param column_indices indices of leaf nodes in parquet schema tree. Appearing ordering
 ///                       matters for the converted schema. Repeated indices are ignored
 ///                       except for the first one
+/// \param key_value_metadata optional metadata, can be nullptr
 /// \param out the corresponding arrow schema
 /// \return Status::OK() on a successful conversion.
 ::arrow::Status PARQUET_EXPORT FromParquetSchema(const SchemaDescriptor* parquet_schema,
-    const std::vector<int>& column_indices, std::shared_ptr<::arrow::Schema>* out);
+    const std::vector<int>& column_indices,
+    const std::shared_ptr<const KeyValueMetadata>& key_value_metadata,
+    std::shared_ptr<::arrow::Schema>* out);
 
-::arrow::Status PARQUET_EXPORT FromParquetSchema(
-    const SchemaDescriptor* parquet_schema, std::shared_ptr<::arrow::Schema>* out);
+::arrow::Status PARQUET_EXPORT FromParquetSchema(const SchemaDescriptor* parquet_schema,
+    const std::shared_ptr<const KeyValueMetadata>& key_value_metadata,
+    std::shared_ptr<::arrow::Schema>* out);
```

Comment:
Interface question: 
I think it's better to keep the previous exported interface.
```
::arrow::Status PARQUET_EXPORT FromParquetSchema(
      const SchemaDescriptor* parquet_schema, std::shared_ptr<::arrow::Schema>* out);
```

so it doesn't break user's code and if user doesn't care about key_value_metadata, he can just ignore them.

what do you think? @wesm